civit_image_downloader.py

```
@@ -437,11 +440,17 @@ async def mark_image_as_downloaded(self, image_id: str, image_path: str, quality
-        wait=wait_random_exponential(multiplier=1, max=10),
-        retry=retry_if_exception(should_retry_exception),
-        before_sleep=before_sleep_log(retry_logger, logging.WARNING)
+    #@retry( # Static decorator using startup retry count
+    #    stop=stop_after_attempt(1 + parse_arguments().retries),
+    #    wait=wait_random_exponential(multiplier=1, max=10),
+    #    retry=retry_if_exception(should_retry_exception),
+    #    before_sleep=before_sleep_log(retry_logger, logging.WARNING)
+    #)
+    @retry(
+        # Only retry if the error is specifically a 429
+        retry=retry_if_exception(lambda e: isinstance(e, requests.HTTPError) and e.response.status_code == 429),
```

Ruff (0.14.10)

Undefined name `requests`

(F821)